tests/rest/client/v2_alpha/test_register.py

```diff
@@ -305,6 +308,73 @@ def test_request_token_existing_email_inhibit_error(self):
 
         self.assertIsNotNone(channel.json_body.get("sid"))
 
+    @unittest.override_config(
+        {
+            "account_threepid_delegates": {
+                "email": "https://id_server",
+                "msisdn": "https://id_server",
+                "registration_only": True,
+            },
+            "public_baseurl": "https://test_server",
+        }
+    )
+    def test_threepid_delegate_registration_only(self):
+        email = "kermit@example.com"
+
+        user_id = self.register_user("kermit", "monkey")
+
+        self.delegated_attempts = 0
+        self.local_attempts = 0
+
+        async def requestEmailToken(
+            id_server, email, client_secret, send_attempt, next_link=None,
+        ):
+            self.delegated_attempts += 1
+            return {"sid": 1234}
+
+        async def send_threepid_validation(
+            email_address, client_secret, send_attempt, send_email_func, next_link=None,
+        ):
+            self.local_attempts += 1
+            return 1234
+
+        identity_handler = self.hs.get_handlers().identity_handler
+        identity_handler.requestEmailToken = Mock(side_effect=requestEmailToken)
+        identity_handler.send_threepid_validation = Mock(
+            side_effect=send_threepid_validation,
+        )
+
+        body = {"client_secret": "somesecret", "email": email, "send_attempt": 0}
+        request, channel = self.make_request(
+            "POST", "/account/3pid/email/requestToken", body,
```

Comment:
These aren't the registration APIs, so I wouldn't expect them to get delegated? These are adding a 3PID to your account

Reply:
As specified in the config doc, `registration_only` refers to registering a 3PID, not registering an account. I agree the wording around this feature is very confusing, but fixing it isn't part of this PR's scope imho.

Reply:
Oh, ok. I think we still should be very careful with the comments and naming of the new option to make that clear. Maybe something like `adding_account_threepid_only`, with a comment that this also does not apply to the registration flows.

Incidentally, I don't think I see why you wouldn't want to do this for registration too?

Reply:
> Incidentally, I don't think I see why you wouldn't want to do this for registration too?

You mean adding a 3PID when registering vs adding a 3PID at another time? I think it might be confusing if Synapse isn't doing the same thing depending on how and when you add an address.

Reply:
I've renamed the config flag in https://github.com/matrix-org/synapse/pull/7611/commits/ff1492ee610ee5f348f035a1f8b9f41789ababbc